llvm/tools/libRemoteCachingService/RemoteCachingService.cpp

```diff
@@ -0,0 +1,34 @@
+//===- RemoteCachingService.cpp ---------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm-c/cas-plugin.h"
+#include "llvm/RemoteCachingService/RemoteCachingService.h"
+#include "llvm/Support/CBindingWrapping.h"
+
+using namespace llvm;
+using namespace llvm::cas;
+
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(CASDB, casdb_t)
+
+static std::string sLastErrorString;
+
+casdb_t llvm_create_plugin_cas(int argc, const char *argv[]) {
+  // If no argument was passed, use default path. Ignore other args.
+  std::string Path = argc == 0 ? "/tmp/compilation-cache-daemon" : argv[1];
```

Comment:
nit: Using `confstr(_CS_DARWIN_USER_TEMP_DIR)` instead of `/tmp` often provides better security.

Reply:
Haha, no. Socket path length is just above 100 char and your system USER_TEMP_DIR can be longer than that (which is the case for our CI that decide to put it under the jenkins build directory).